src/DotNetWorker/Http/HttpRequestData.cs

```diff
@@ -1,18 +1,33 @@
-// Copyright (c) .NET Foundation. All rights reserved.
+﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Security.Claims;
+using Google.Protobuf.WellKnownTypes;
 
 namespace Microsoft.Azure.Functions.Worker
 {
-    public abstract class HttpRequestData
+    // <inheritdoc/>
+    public abstract class HttpRequestData : IHttpRequestData
     {
+        // <inheritdoc/>
+        public abstract ReadOnlyMemory<byte>? Body { get; }
+
+        // <inheritdoc/>
         public abstract IImmutableDictionary<string, string> Headers { get; }
 
-        public abstract string Body { get; }
+        // <inheritdoc/>
+        public abstract IReadOnlyCollection<IHttpCookie> Cookies { get; }
+
+        // <inheritdoc/>
+        public abstract Uri Url { get; }
 
-        public abstract IImmutableDictionary<string, string> Params { get; }
```

Comment:
I think it makes sense to bring back "Params"? (And not expose trigger_metadata in the future)

Reply:
After the discussion we had, I'm inclined to delay this for now to make sure this is the right model. Thoughts?